spec/graphql/queries/dogs_spec.rb

```diff
@@ -1,44 +1,118 @@
+    @d4 = create(:dog, user: @u2, breed: 'Rat Terrier', activity_level: 0, birthdate: '2015-03-18', weight: 20)
+    @d5 = create(:dog, user: @u2, breed: 'Malinois', activity_level: 1, birthdate: '2010-07-01', weight: 60)
+    @d6 = create(:dog, user: @u2, breed: 'Tibetan Terrier', activity_level: 2, birthdate: '2003-10-04', weight: 100)
+  end
+
+  it 'returns all dogs' do
+    query = "query { dogs { #{dog_type_attributes} user { #{user_type_attributes} } photos { #{photo_type_attributes} }}}"
 
     post '/graphql', params: { query: query }
-    json = JSON.parse(response.body)
-    data = json['data']['dogs']
 
-    expect(data.count).to eq(3)
+    json = JSON.parse(response.body, symbolize_names: true)
+    data = json[:data][:dogs]
+
+    expect(data.count).to eq(6)
 
     first_gql_dog = data.first
-    compare_gql_and_db_dogs(first_gql_dog, dog11)
+    compare_gql_and_db_dogs(first_gql_dog, @d1)
 
-    gql_user_of_first_dog = first_gql_dog['user']
-    compare_gql_and_db_users(gql_user_of_first_dog, user1)
+    gql_user_of_first_dog = first_gql_dog[:user]
+    compare_gql_and_db_users(gql_user_of_first_dog, @u1)
 
-    gql_photos = first_gql_dog['photos']
+    gql_photos = first_gql_dog[:photos]
     expect(gql_photos.count).to eq(1)
-    compare_gql_and_db_photos(gql_photos.first, photo)
+    compare_gql_and_db_photos(gql_photos.first, @p1)
+  end
+
+  it 'returns all dogs filtered by activity level' do
+    query = "query { dogs(activityLevelRange: [0, 1]) { #{dog_type_attributes} user { #{user_type_attributes} }}}"
+
+    post '/graphql', params: { query: query }
+
+    json = JSON.parse(response.body, symbolize_names: true)
+    data = json[:data][:dogs]
+
+    expect(data.count).to eq(4)
+
+    expect(data[0][:activityLevel]).to eq(0).or eq(1)
+    expect(data[1][:activityLevel]).to eq(0).or eq(1)
+    expect(data[2][:activityLevel]).to eq(0).or eq(1)
+    expect(data[3][:activityLevel]).to eq(0).or eq(1)
+  end
+
+  it 'returns all dogs filtered by weight' do
+    query = "query { dogs(weightRange: [60, 100]) { #{dog_type_attributes} user { #{user_type_attributes} }}}"
+
+    post '/graphql', params: { query: query }
+
+    json = JSON.parse(response.body, symbolize_names: true)
+    data = json[:data][:dogs]
+
+    expect(data.count).to eq(4)
+
+    expect(data[0][:weight]).to be_between(60, 100).inclusive
+    expect(data[1][:weight]).to be_between(60, 100).inclusive
+    expect(data[2][:weight]).to be_between(60, 100).inclusive
+    expect(data[3][:weight]).to be_between(60, 100).inclusive
   end
 
-  def query
-    <<~GQL
-      query {
-        dogs {
-          #{dog_type_attributes}
-          user {
-            #{user_type_attributes}
-          }
-          photos {
-            #{photo_type_attributes}
-          }
-        }
-      }
-    GQL
+  it 'returns all dogs filtered by breed' do
+    query = "query { dogs(breed: [\"Rat Terrier\", \"Tibetan Terrier\"]) { #{dog_type_attributes} user { #{user_type_attributes} }}}"
```

Comment:
```suggestion
    query = "query {
      dogs(breed: [\"Rat Terrier\", \"Tibetan Terrier\"]) {
        #{dog_type_attributes}
        user {
          #{user_type_attributes}
        }
      }
    }"
```

Reply:
Looks like the arguments to the `dogs` query is the only thing that changes between all these tests -- opportunity to DRY up.